app/models/compute_resources/foreman/model/vmware.rb

```diff
@@ -523,6 +527,7 @@ def create_vm(args = { })
       else
         vm = new_vm(args)
         vm.firmware = 'bios' if vm.firmware == 'automatic'
+        raise errors.full_messages.join(', ') unless errors.empty?
```

Comment:
The first check `errors.empty?` is for the test_connection. However, in this case, we're checking if any errors occurred while calling the new_vm method. If the firmware is set to BIOS and the virtual TPM is set to true, we will get an error: https://github.com/theforeman/foreman/blob/0c60e217451551ed27ba87a9b33178200101e7bc/app/models/compute_resources/foreman/model/vmware.rb#L490

I chose this approach to avoid disrupting the entire process. If the user is using the UI, I want to retain the form with the values they have set and only display this specific error message.